components/engine_rocks/src/raft_engine.rs

```diff
@@ -13,6 +18,12 @@ use tikv_util::{box_err, box_try};
 
 const RAFT_LOG_MULTI_GET_CNT: u64 = 8;
 
+lazy_static! {
+    pub static ref RAFT_LOG_GC_ON_COMPACTION: AtomicBool = AtomicBool::new(false);
+    pub static ref RAFT_LOG_GC_INDEXES: Arc<RwLock<HashMap<u64, u64>>> =
```

Comment:
I'm not sure is there any performance issue here. But I prefer this:
```
impl AdminObserver {
    fn on_create_region(&self) {
        let truncate_index = Arc::new(AtomicU64::new(0));
        RAFT_LOG_GC_INDEXES.write().unwrap().insert(region_id, truncate_index.clone());
        peer.peer_storage.truncate_index = truncate_index;
    }
    fn on_remove_region(&self) {
         ......
    }
}
```
So that the the write lock of the hashmap is only required when creating/removing a Region on a store. Update truncate index for a Region requires no lock.

Reply:
That's a good idea. I will check `AdminObserver` and implement it.